src/Models/DataType.php

```diff
@@ -141,7 +142,17 @@ public function fieldOptions()
     {
         $table = $this->name;
 
-        $fieldOptions = SchemaManager::describeTable($table);
+        // Get ordered BREAD fields
+        $orderedFields = $this->rows()->pluck('field')->toArray();
+
+        $_fieldOptions = SchemaManager::describeTable($table)->toArray();
+
+        $fieldOptions = [];
+        $f_size = count($orderedFields);
+        for ($i = 0; $i < $f_size; $i++) {
```

Comment:
The problem is that we iterate on $orderedFields which do not contains new fields. We should work on $_fieldOptions instead.
I've added a PR for that https://github.com/the-control-group/voyager/pull/1210